Lib/subprocess.py

```diff
@@ -1314,6 +1314,8 @@ def _internal_poll(self, _deadstate=None,
             if self.returncode is None:
                 if _WaitForSingleObject(self._handle, 0) == _WAIT_OBJECT_0:
                     self.returncode = _GetExitCodeProcess(self._handle)
+                    self._handle.Close()
+                    self._handle = None
```

Comment:
`__exit__` calls `wait`, which calls `_wait`, which does not call `_internal_poll`. This should be implemented in `_wait` first and foremost. It can also be implemented here, or modify `_internal_poll` to call `_wait(0)`.

Reply:
Oh right, I forgot wait() and terminate(). It's now fixed.

I don't want to bother with a timeout exception in _internal_poll(), so I didn't modify it to reuse wait().